tests/buildTools/src/test/scala/tests/BaseBuildToolSuite.scala

```diff
@@ -130,3 +148,56 @@ abstract class BaseBuildToolSuite extends MopedSuite(ScipJava.app) {
   }
 
 }
+
+object BaseBuildToolSuite {
+  lazy val externalJavaVersion: Int = {
+    val tmpDir = os.temp.dir()
+    var version = Option.empty[String]
+    try {
+      os.write(tmpDir / "PrintJavaVersion.java", PrintJavaVersion)
+
+      os.proc("javac", "PrintJavaVersion.java").call(cwd = tmpDir)
+
+      version = Some(
+        os.proc("java", "PrintJavaVersion").call(cwd = tmpDir).out.text()
+      )
+    } finally {
+      os.remove.all(tmpDir)
+    }
+
+    version
+      .map(parseJavaVersion)
+      .getOrElse(sys.error("Failed to detect external JDK version"))
+  }
+
+  private def parseJavaVersion(raw: String) = {
+    val prop = raw.takeWhile(c => c.isDigit || c == '.')
+
+    val segments = prop.split("\\.").toList
+
+    segments match {
+      // Java 1.6 - 1.8
+      case "1" :: lessThan8 :: _ :: Nil =>
+        lessThan8.toInt
+      // Java 17.0.1, 11.0.20.1, ..
+      case modern :: _ :: _ :: rest =>
+        modern.toInt
+      // Java 12
+      case modern :: Nil =>
+        modern.toInt
+      case other =>
+        sys.error(
+          s"Cannot process [java.version] property, unknown format: [$raw]"
+        )
+    }
+  }
+
+  private val PrintJavaVersion = """
+      public class PrintJavaVersion {
+        public static void main(String[] args) {
+          System.out.print(System.getProperty("java.version"));
+        }
+      }
```

Comment:
This is a more reliable way of getting the version out of Java, because parsing `java -version` is more difficult due to variability in how that information is presented depending on version _and_ vendor.

Logic and mechanism were lifted from https://github.com/sourcegraph/sbt-sourcegraph/pull/78